doc/source/skips/4-transition-to-v2.rst

```diff
@@ -0,0 +1,255 @@
+  number to 2.0-dev.
+- There will be *no* scikit-image package on PyPI with version 2.0. Users who
+  ``pip install scikit-image`` will always get the 1.0 version of the package.
+- After consensus has been reached on the new API, skimage2 will be released.
+- scikit-image 1.0.x will receive critical bug fixes for an unspecified period
+  of time, depending on the severity of the bug and the amount of effort
+  involved.
+
+Backward compatibility
+----------------------
+
+This proposal breaks backward compatibility in numerous places in the library.
+However, it does so in a new namespace, so that this proposal does not raise
+backward compatibilty concerns for our users. That said, the authors will
+attempt to limit the number of backward incompatible changes to those likely to
+substantially improve the overall user experience. It is anticipated that
+porting `skimage` code to `skimage2` will be a straightforward process
+and we will publish a user guide for making the transition by the time of
+the `skimage2` release.
+
+Alternatives
+------------
+
+Releasing the new API in the same package using semantic versioning
+...................................................................
+
+This is `SKIP-3 <skip_3_transition_v1>_`, which was rejected after discussion
+with the community.
+
+Continuous deprecation over multiple versions
+.............................................
+
+This transition could occur gradually over many versions. For example, for
+functions automatically converting and rescaling float inputs, we could add a
+``preserve_range`` keyword argument that would initially default to False, but
+the default value of False would be deprecated, with users getting a warning to
+switch to True. After the switch, we could (optionally) deprecate the
+argument, arriving, after a further two releases, at the same place:
+scikit-image no longer rescales data automatically, there are no
+unnecessary keyword arguments lingering all over the API.
+
+Of course, this kind of operation would have to be done simultaneously over all
+of the above proposed changes.
+
+Ultimately, the core team felt that this approach generates more work for both
+the scikit-image developers and the developers of downstream libraries, for
+dubious benefit: ultimately, later versions of scikit-image will still be
+incompatible with prior versions, although over a longer time scale.
+
+A single package containing both versions
+.........................................
+
+Since the import name is changing, it would be possible to make a single
+package with both the ``skimage`` and ``skimage2`` namespaces shipping
+together, at least for some time. This option is attractive but it implies
+longer-term maintenance of the 1.0 namespace, for which we might lack
+maintainer time, or a long deprecation cycle for the 1.0 namespace, which would
+ultimately result in a lot of unhappy users getting deprecation messages from
+their scikit-image use.
+
```

Comment:
I do like separate packages better, but here are two concrete examples that did go with this combined package approach:
- OpenCV had both `cv` and `cv2` namespaces available in the same package for a little while.
- `imageio` recently introduced `imagio.v2` and `imageio.v3` with the plan that the default `imageio` namespace will switch from v2 to v3 at a future date.

Reply:
[BeautifulSoup](https://beautiful-soup-4.readthedocs.io/en/latest/#porting-code-to-bs4) is also a good example, I guess.